multi_partner_learning.py

```diff
@@ -487,6 +528,38 @@ def init_with_agg_models(self):
         for partner in self.partners_list:
             partners_model_list.append(self.build_model_from_weights(self.aggregate_model_weights()))
         return partners_model_list
+    
+    
+    def init_with_previous_learned_model(self):
+        """Return a new model aggregating models from model_list"""
+        
+        previous_model = self.generate_new_model()
+        previous_model.load_weights(self.init_model_from)
+        previous_model.compile(
+            loss=keras.losses.categorical_crossentropy,
+            optimizer="adam",
+            metrics=["accuracy"],
+        )
+
+        return previous_model
+
+
+    def init_with_previous_learned_models(self):
+        """Return a list with the aggregated model duplicated for each partner"""
+
+        previous_model = self.generate_new_model()
+        previous_model.load_weights(self.init_model_from)
+        previous_model.compile(
+            loss=keras.losses.categorical_crossentropy,
+            optimizer="adam",
+            metrics=["accuracy"],
+        )
+        
+        partners_model_list = []
+        for partner in self.partners_list:
+            partners_model_list.append(previous_model)
+        return partners_model_list
```

Comment:
There is a catch here. As it is, you create different pointers to the same model. To have different models for each partner, you should create one for each.